target_chains/solana/sdk/js/pyth_solana_receiver/src/vaa.ts

```diff
@@ -54,81 +76,250 @@ export function trimSignatures(
+        computeUnits: WRITE_ENCODED_VAA_COMPUTE_BUDGET,
+      },
+      {
+        instruction: await wormhole.methods
+          .verifyEncodedVaaV1()
+          .accounts({
+            guardianSet: getGuardianSetPda(
+              getGuardianSetIndex(vaa),
+              wormhole.programId
+            ),
+            draftVaa: encodedVaaKeypair.publicKey,
+          })
+          .instruction(),
+        signers: [],
+        computeUnits: VERIFY_ENCODED_VAA_COMPUTE_BUDGET,
+      },
+    ];
+
+  // Close instructions
+  const closeInstructions: InstructionWithEphemeralSigners[] = [
     {
       instruction: await wormhole.methods
-        .writeEncodedVaa({
-          index: VAA_SPLIT_INDEX,
-          data: vaa.subarray(VAA_SPLIT_INDEX),
-        })
-        .accounts({
-          draftVaa,
-        })
+        .closeEncodedVaa()
+        .accounts({ encodedVaa: encodedVaaKeypair.publicKey })
         .instruction(),
       signers: [],
-      computeUnits: WRITE_ENCODED_VAA_COMPUTE_BUDGET,
+      computeUnits: CLOSE_ENCODED_VAA_COMPUTE_BUDGET,
     },
   ];
+
+  return {
+    initInstructions,
+    writeFirstPartInstructions,
+    writeSecondPartAndVerifyInstructions,
+    closeInstructions,
+    encodedVaaAddress: encodedVaaKeypair.publicKey,
+  };
+}
+
+/**
+ * Build instructions to post a single VAA to the Wormhole program.
+ * The instructions are packed efficiently into 2 transactions:
+ * - TX1: Create, init the encoded VAA account and write the first part of the VAA
+ * - TX2: Write the second part of the VAA and verify it
+ *
+ * @param wormhole - The Wormhole program instance
+ * @param vaa - The VAA buffer to post
+ * @returns {Object} Result containing:
+ *   - encodedVaaAddress: Public key of the encoded VAA account
+ *   - postInstructions: Instructions to post and verify the VAA
+ *   - closeInstructions: Instructions to close the encoded VAA account and recover rent
+ */
+export async function buildPostEncodedVaaInstructions(
+  wormhole: Program<WormholeCoreBridgeSolana>,
+  vaa: Buffer
+): Promise<{
+  encodedVaaAddress: PublicKey;
+  postInstructions: InstructionWithEphemeralSigners[];
+  closeInstructions: InstructionWithEphemeralSigners[];
+}> {
+  const groups = await generateVaaInstructionGroups(wormhole, vaa);
+
+  // Pack instructions for optimal 2-transaction pattern:
+  // TX1: init + first write
+  // TX2: second write + verify
+  return {
+    encodedVaaAddress: groups.encodedVaaAddress,
+    postInstructions: [
+      ...groups.initInstructions,
+      ...groups.writeFirstPartInstructions,
+      ...groups.writeSecondPartAndVerifyInstructions,
+    ],
+    closeInstructions: groups.closeInstructions,
+  };
+}
+
+/**
+ * Build instructions to post two VAAs for TWAP (Time-Weighted Average Price) calculations,
+ * optimized for 3 transactions. This is specifically designed for posting start and end
+ * accumulator update VAAs efficiently.
+ * The instructions are packed into 3 transactions:
```

Comment:
Nitpick: this function doesn't handle the packing.
Maybe you can say that this function splits the vaa writes the `writeEncodedVaa` in multiple instructions with the intention that these instructions can be packed in that way.